satpy/readers/gld360_ualf2.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Gld360Ualf2 reader.
+
+Vaisala Global Lightning Dataset GLD360 is data as a service
+that provides real-time lightning data for accurate and early
+detection and tracking of severe weather. The data provided is
+generated by a Vaisala owned and operated world-wide lightning
+detection sensor network.
+
+References:
+- [GLD360] https://www.vaisala.com/en/products/data-subscriptions-and-reports/data-sets/gld360
+
+"""
+
+import logging
+from datetime import timedelta
+
+import dask.dataframe as dd
+import numpy as np
+import xarray as xr
+
+from satpy.readers.file_handlers import BaseFileHandler
+
+logger = logging.getLogger(__name__)
+
+
+class Gld360Ualf2FileHandler(BaseFileHandler):
+    """ASCII reader for Vaisala GDL360 ualf2 data."""
+
+    def __init__(self, filename, filename_info, filetype_info):
+        """Initialize Vaisala Gld360Ualf2FileHandler."""
+        super(Gld360Ualf2FileHandler, self).__init__(filename, filename_info, filetype_info)
+
+        def pad_nanoseconds(nanoseconds):
+            """Read ns values for less than 0.1s correctly (these are not zero-padded in the input files)."""
+            return str(nanoseconds).zfill(9)
+
+        names = [
+            'ualf_record_type',
+            'network_type',
+            'year',
+            'month',
+            'day',
+            'hour',
+            'minute',
+            'second',
+            'nanosecond',
+            'latitude',
+            'longitude',
+            'altitude',
+            'altitude_uncertainty',
+            'peak_current',
+            'vhf_range',
+            'multiplicity_flash',
+            'cloud_pulse_count',
+            'number_of_sensors',
+            'degree_freedom_for_location',
+            'error_ellipse_angle',
+            'error_ellipse_max_axis_length',
+            'error_ellipse_min_axis_length',
+            'chi_squared_value_location_optimization',
+            'wave_form_rise_time',
+            'wave_form_peak_to_zero_time',
+            'wave_form_max_rate_of_rise',
+            'cloud_indicator',
+            'angle_indicator',
+            'signal_indicator',
+            'timing_indicator',
+        ]
+
+        dtype = {
```

Comment:
same as above for the `names`

Reply:
dtype `keys` here are duplicates of the `names`-list above. maybe the `names`-list can be removed altogether?